pkg/k8s/labels.go

```diff
@@ -57,8 +59,39 @@ const (
 	ProxyVersionAnnotation = "conduit.io/proxy-version"
 )
 
+var proxyLabels = []string{
+	ProxyDeploymentLabel,
+	ProxyReplicationControllerLabel,
+	ProxyReplicaSetLabel,
+	ProxyJobLabel,
+	ProxyDaemonSetLabel,
+}
+
 // CreatedByAnnotationValue returns the value associated with
 // CreatedByAnnotation.
 func CreatedByAnnotationValue() string {
 	return fmt.Sprintf("conduit/cli %s", version.Version)
 }
+
+// GetOwnerLabels returns the set of prometheus owner labels that can be
+// extracted from the proxy labels that have been added to an injected
+// kubernetes resource
+func GetOwnerLabels(objectMeta meta.ObjectMeta) map[string]string {
+	labels := make(map[string]string)
+	for _, label := range proxyLabels {
+		if labelValue, ok := objectMeta.Labels[label]; ok {
+			labels[toOwnerLabel(label)] = labelValue
+		}
+	}
+	return labels
+}
+
+// toOwnerLabel converts a proxy label to a prometheus label, following the
+// relabel conventions from the prometheus scrape config file
+func toOwnerLabel(proxyLabel string) string {
```

Comment:
I think this is good enough for now, but I had expected that the destination service would derive ownership information from Kubernetes, not from labels injected into the pods. This is particularly relevant for things like ReplicaSets, where Conduit has no knowledge of this at deployment inject time, so those labels will never be set by Conduit.

Related, I am re-introducing a `getPodOwner` method into public-api, cribbed from the telemetry service, that will probably do what we need:
https://github.com/runconduit/conduit/blob/813aa0fddcf5a01ea673a7204ebe2da6932ef4a7/controller/api/public/grpc_server.go#L529-L546

```go
func (s *grpcServer) getPodOwner(pod *k8sV1.Pod) (string, error) {
	namespace := pod.Namespace
	if len(pod.GetOwnerReferences()) == 0 {
		return "", fmt.Errorf("Pod %s has no owner", pod.Name)
	}
	parent := pod.GetOwnerReferences()[0]
	if parent.Kind == "ReplicaSet" {
		rs, err := s.replicaSetLister.GetPodReplicaSets(pod)
		if err != nil {
			return "", err
		}
		if len(rs) == 0 || len(rs[0].GetOwnerReferences()) == 0 {
			return namespace + "/" + parent.Name, nil
		}
		return namespace + "/" + rs[0].GetOwnerReferences()[0].Name, nil
	}
	return namespace + "/" + parent.Name, nil
}
```

Reply:
Ok, this makes sense -- I agree that it would be better to get ownership info directly from kubernetes if available. We should create an issue to do this as a follow-up.